[Search Sessions] Make search session indicator UI opt-in

Dosant opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/88442

Previously the idea was that the search session indicator won't appear if an app doesn't use the search session. But recently Lens started using them without completing full integration https://github.com/elastic/kibana/pull/86297. If to enable `xpack.data_enhanced.search.sessions.enabled` in master, then Lens show search session indicator, but sending session to background doesn't properly work. 

This pr changes the underlying logic of session service to show session indicator only app explicitly asked for that. 

Also, I noticed, that we checked for per-app capabilities inside the data plugin component. I moved it outside and so now apps can configure search session indicator UI. In this case by setting it to disabled state.


### What to test

The only user-facing difference comparing to master is that the search session indicator shouldn't appear in the Lens app.




### Checklist


- [x] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios

### For maintainers

- [x] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/)
* Commit: c93ef6a1f82801fcbd2d4240437d4825823127d6

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/7c9b11476dddbfdc27acefd27741744d3c9f202e) | [after](https://github.com/elastic/kibana/commit/c93ef6a1f82801fcbd2d4240437d4825823127d6) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/7c9b11476dddbfdc27acefd27741744d3c9f202e) | [after](https://github.com/elastic/kibana/commit/c93ef6a1f82801fcbd2d4240437d4825823127d6) | diff |
| --- | --- | --- | --- |
| `dashboard` | 200.3KB | 200.5KB | +276.0B |
| `dataEnhanced` | 27.4KB | 27.4KB | -1.0B |
| `discover` | 511.3KB | 511.4KB | +142.0B |
| total |  |  | +417.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/7c9b11476dddbfdc27acefd27741744d3c9f202e) | [after](https://github.com/elastic/kibana/commit/c93ef6a1f82801fcbd2d4240437d4825823127d6) | diff |
| --- | --- | --- | --- |
| `dashboard` | 334.2KB | 334.5KB | +325.0B |
| `data` | 1003.8KB | 1005.4KB | +1.6KB |
| `dataEnhanced` | 38.9KB | 38.5KB | -434.0B |
| total |  |  | +1.5KB |

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/","number":"99642","commit":"c93ef6a1f82801fcbd2d4240437d4825823127d6"}]}
PIPELINE-->

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServices)

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/)
* Commit: 4d041db4fa476d5161f3bdab1c5c298b571c927a

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/37cc7af49d7f1f81f8e6265d05dc1fd355b0ca24) | [after](https://github.com/elastic/kibana/commit/4d041db4fa476d5161f3bdab1c5c298b571c927a) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/37cc7af49d7f1f81f8e6265d05dc1fd355b0ca24) | [after](https://github.com/elastic/kibana/commit/4d041db4fa476d5161f3bdab1c5c298b571c927a) | diff |
| --- | --- | --- | --- |
| `dashboard` | 200.3KB | 200.5KB | +276.0B |
| `dataEnhanced` | 27.4KB | 27.4KB | -1.0B |
| `discover` | 511.3KB | 511.4KB | +142.0B |
| total |  |  | +417.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/37cc7af49d7f1f81f8e6265d05dc1fd355b0ca24) | [after](https://github.com/elastic/kibana/commit/4d041db4fa476d5161f3bdab1c5c298b571c927a) | diff |
| --- | --- | --- | --- |
| `dashboard` | 334.2KB | 334.5KB | +325.0B |
| `data` | 1003.8KB | 1005.4KB | +1.6KB |
| `dataEnhanced` | 38.9KB | 38.5KB | -434.0B |
| total |  |  | +1.5KB |

### History
* :green_heart: [Build #99934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/) succeeded c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f
* :green_heart: [Build #99642](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/) succeeded c93ef6a1f82801fcbd2d4240437d4825823127d6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/","number":"99642","commit":"c93ef6a1f82801fcbd2d4240437d4825823127d6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/","number":"99934","commit":"c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/","number":"99951","commit":"4d041db4fa476d5161f3bdab1c5c298b571c927a"}]}
PIPELINE-->

Dosant commented:
@elasticmachine merge upstream

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/)
* Commit: 3f3f71bb695cada935ff0c6c1c3eef18be8c083b
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/execution/node/365/log)

### History
* :green_heart: [Build #99951](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/) succeeded 4d041db4fa476d5161f3bdab1c5c298b571c927a
* :green_heart: [Build #99934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/) succeeded c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f
* :green_heart: [Build #99642](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/) succeeded c93ef6a1f82801fcbd2d4240437d4825823127d6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/","number":"99642","commit":"c93ef6a1f82801fcbd2d4240437d4825823127d6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/","number":"99934","commit":"c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/","number":"99951","commit":"4d041db4fa476d5161f3bdab1c5c298b571c927a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/","number":"100278","commit":"3f3f71bb695cada935ff0c6c1c3eef18be8c083b"}]}
PIPELINE-->

lizozom commented on src/plugins/data/public/search/session/i18n.ts:
```diff
@@ -0,0 +1,20 @@
+export const noSearchSessionStorageCapabilityMessage = i18n.translate(
+  'data.searchSessionIndicator.noCapability',
+  {
+    defaultMessage: "You don't have permissions to send to background.",
```
```suggestion
    defaultMessage: "You don't have permissions to create search sessions.",
```

lizozom commented on src/plugins/dashboard/public/application/hooks/use_dashboard_state_manager.ts:
```diff
@@ -159,13 +160,22 @@ export const useDashboardStateManager = (
+            ? { disabled: false }
+            : {
+                disabled: true,
+                reasonText: noSearchSessionStorageCapabilityMessage,
```
Could we please keep the reason text inside the plugin?
I think it's a too verbose API.

Dosant replied:
current API doesn't limit disabling the UI for other reasons than not enough permissions. 
In case some other app has other app-specific use cases where they want to disable storing a session, they can still use this API by providing different explanation text.

For example, in a dashboard, I'd use this to disable storing a session when in edit mode.

lizozom replied:
:+1:

lizozom commented on src/plugins/data/public/search/session/session_service.ts:
```diff
@@ -269,4 +274,34 @@ export class SessionService {
+   * @param searchSessionInfoProvider - info provider for saving a search session
+   * @param searchSessionIndicatorUiConfig - config for "Search session indicator" UI
+   */
+  public setupStorage<ID extends UrlGeneratorId = UrlGeneratorId>(
```
I think `enable` would be a better name for this. What do you say?

lizozom reviewed (commented):
Overall looks great.
Maybe we could add this to the `sessions_in_space` tests?
Should be simple enough to create a user that doesn't have this permission, and see that the component is properly disabled.

lizozom commented on src/plugins/data/public/search/session/session_service.test.ts:
```diff
@@ -156,4 +156,51 @@ describe('Session service', () => {
     expect(sessionService.isCurrentSession(sessionId)).toBeTruthy();
   });
+
+  test('setupStorage() enables storage capabilities', async () => {
```
To imitate someone using this service's api (and not via UI)
Can you add a test for calling the `save` api, when `setupStorage` wasn't called?
It should fail, right?

Also, might be nice to test what happens if `getUrlGeneratorData` throws and error while saving.

Dosant replied:
> Can you add a test for calling the save api, when setupStorage wasn't called?

There is one: 
https://github.com/elastic/kibana/blob/6cd417744d63642dc35903b0fec42ccf0b99f880/src/plugins/data/public/search/session/session_service.test.ts#L162

Or did you mean something else?

Dosant replied:
> Also, might be nice to test what happens if getUrlGeneratorData throws and error while saving.

Added this one

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/)
* Commit: 6cd417744d63642dc35903b0fec42ccf0b99f880

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/dfd96d62c42207dd4b700a63548428aaded282af) | [after](https://github.com/elastic/kibana/commit/6cd417744d63642dc35903b0fec42ccf0b99f880) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/dfd96d62c42207dd4b700a63548428aaded282af) | [after](https://github.com/elastic/kibana/commit/6cd417744d63642dc35903b0fec42ccf0b99f880) | diff |
| --- | --- | --- | --- |
| `dashboard` | 200.3KB | 200.5KB | +276.0B |
| `discover` | 511.3KB | 511.4KB | +142.0B |
| total |  |  | +418.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/dfd96d62c42207dd4b700a63548428aaded282af) | [after](https://github.com/elastic/kibana/commit/6cd417744d63642dc35903b0fec42ccf0b99f880) | diff |
| --- | --- | --- | --- |
| `dashboard` | 334.2KB | 334.5KB | +325.0B |
| `data` | 1010.1KB | 1011.7KB | +1.6KB |
| `dataEnhanced` | 41.8KB | 41.3KB | -434.0B |
| total |  |  | +1.5KB |

### History
* :broken_heart: [Build #100278](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/) failed 3f3f71bb695cada935ff0c6c1c3eef18be8c083b
* :green_heart: [Build #99951](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/) succeeded 4d041db4fa476d5161f3bdab1c5c298b571c927a
* :green_heart: [Build #99934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/) succeeded c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f
* :green_heart: [Build #99642](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/) succeeded c93ef6a1f82801fcbd2d4240437d4825823127d6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/","number":"99642","commit":"c93ef6a1f82801fcbd2d4240437d4825823127d6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/","number":"99934","commit":"c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/","number":"99951","commit":"4d041db4fa476d5161f3bdab1c5c298b571c927a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/","number":"100278","commit":"3f3f71bb695cada935ff0c6c1c3eef18be8c083b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/","number":"100309","commit":"6cd417744d63642dc35903b0fec42ccf0b99f880"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/)
* Commit: d01cade01ac63d0cf2ee55843709f69cf1faa548
* This comment will update when the build is complete

### Failed CI Steps
* [X-Pack Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/execution/node/337/log)

### History
* :green_heart: [Build #100309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/) succeeded 6cd417744d63642dc35903b0fec42ccf0b99f880
* :broken_heart: [Build #100278](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/) failed 3f3f71bb695cada935ff0c6c1c3eef18be8c083b
* :green_heart: [Build #99951](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/) succeeded 4d041db4fa476d5161f3bdab1c5c298b571c927a
* :green_heart: [Build #99934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/) succeeded c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f
* :green_heart: [Build #99642](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/) succeeded c93ef6a1f82801fcbd2d4240437d4825823127d6

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99642/","number":"99642","commit":"c93ef6a1f82801fcbd2d4240437d4825823127d6"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/","number":"99934","commit":"c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/","number":"99951","commit":"4d041db4fa476d5161f3bdab1c5c298b571c927a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/","number":"100278","commit":"3f3f71bb695cada935ff0c6c1c3eef18be8c083b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/","number":"100309","commit":"6cd417744d63642dc35903b0fec42ccf0b99f880"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/","number":"100367","commit":"d01cade01ac63d0cf2ee55843709f69cf1faa548"}]}
PIPELINE-->

Dosant commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/)
* Commit: 75eebcaa1811b142c9485a49378a36608df67e54

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/ed811e332def6e19c5e7bcc3ed719394e3da3b72) | [after](https://github.com/elastic/kibana/commit/75eebcaa1811b142c9485a49378a36608df67e54) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/ed811e332def6e19c5e7bcc3ed719394e3da3b72) | [after](https://github.com/elastic/kibana/commit/75eebcaa1811b142c9485a49378a36608df67e54) | diff |
| --- | --- | --- | --- |
| `dashboard` | 200.3KB | 200.5KB | +277.0B |
| `discover` | 511.3KB | 511.4KB | +143.0B |
| total |  |  | +420.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/ed811e332def6e19c5e7bcc3ed719394e3da3b72) | [after](https://github.com/elastic/kibana/commit/75eebcaa1811b142c9485a49378a36608df67e54) | diff |
| --- | --- | --- | --- |
| `dashboard` | 334.2KB | 334.5KB | +325.0B |
| `data` | 1010.1KB | 1011.7KB | +1.6KB |
| `dataEnhanced` | 41.8KB | 41.3KB | -434.0B |
| total |  |  | +1.5KB |

### History
* :broken_heart: [Build #100367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/) failed d01cade01ac63d0cf2ee55843709f69cf1faa548
* :green_heart: [Build #100309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/) succeeded 6cd417744d63642dc35903b0fec42ccf0b99f880
* :broken_heart: [Build #100278](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/) failed 3f3f71bb695cada935ff0c6c1c3eef18be8c083b
* :green_heart: [Build #99951](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/) succeeded 4d041db4fa476d5161f3bdab1c5c298b571c927a
* :green_heart: [Build #99934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/) succeeded c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99934/","number":"99934","commit":"c8e2a8a0ee5c6a0a47fabe1a3ab1bb20322d514f"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/","number":"99951","commit":"4d041db4fa476d5161f3bdab1c5c298b571c927a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/","number":"100278","commit":"3f3f71bb695cada935ff0c6c1c3eef18be8c083b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/","number":"100309","commit":"6cd417744d63642dc35903b0fec42ccf0b99f880"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/","number":"100367","commit":"d01cade01ac63d0cf2ee55843709f69cf1faa548"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/","number":"100393","commit":"75eebcaa1811b142c9485a49378a36608df67e54"}]}
PIPELINE-->

lizozom commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/)
* Commit: 6385ef545d7cbc7cb5e0b343111bf5805d227617

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/8263d47d378315bdcad990620010c8452f3133d1) | [after](https://github.com/elastic/kibana/commit/6385ef545d7cbc7cb5e0b343111bf5805d227617) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/8263d47d378315bdcad990620010c8452f3133d1) | [after](https://github.com/elastic/kibana/commit/6385ef545d7cbc7cb5e0b343111bf5805d227617) | diff |
| --- | --- | --- | --- |
| `dashboard` | 200.3KB | 200.5KB | +277.0B |
| `discover` | 511.3KB | 511.4KB | +143.0B |
| total |  |  | +420.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/8263d47d378315bdcad990620010c8452f3133d1) | [after](https://github.com/elastic/kibana/commit/6385ef545d7cbc7cb5e0b343111bf5805d227617) | diff |
| --- | --- | --- | --- |
| `dashboard` | 334.2KB | 334.5KB | +325.0B |
| `data` | 1013.3KB | 1014.9KB | +1.6KB |
| `dataEnhanced` | 41.8KB | 41.3KB | -434.0B |
| total |  |  | +1.5KB |

### History
* :green_heart: [Build #100393](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/) succeeded 75eebcaa1811b142c9485a49378a36608df67e54
* :broken_heart: [Build #100367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/) failed d01cade01ac63d0cf2ee55843709f69cf1faa548
* :green_heart: [Build #100309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/) succeeded 6cd417744d63642dc35903b0fec42ccf0b99f880
* :broken_heart: [Build #100278](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/) failed 3f3f71bb695cada935ff0c6c1c3eef18be8c083b
* :green_heart: [Build #99951](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/) succeeded 4d041db4fa476d5161f3bdab1c5c298b571c927a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/99951/","number":"99951","commit":"4d041db4fa476d5161f3bdab1c5c298b571c927a"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/","number":"100278","commit":"3f3f71bb695cada935ff0c6c1c3eef18be8c083b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/","number":"100309","commit":"6cd417744d63642dc35903b0fec42ccf0b99f880"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/","number":"100367","commit":"d01cade01ac63d0cf2ee55843709f69cf1faa548"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/","number":"100393","commit":"75eebcaa1811b142c9485a49378a36608df67e54"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/","number":"100597","commit":"6385ef545d7cbc7cb5e0b343111bf5805d227617"}]}
PIPELINE-->

Dosant commented:
@lizozom,
> Maybe we could add this to the sessions_in_space tests?

I added these tests as you suggested.
I also added a lens test checking that there is no search session indicator in the UI.

flash1293 commented:
Requires a deeper look into Discover, but indicator isn't showing up anymore in Lens, which LGTM

lizozom reviewed (approved):
Just tested with and without permissions and seems good!
LGTM

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100646/)
* Commit: a0235f861dd68ae902534a1537aedad74bea50cd

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/8263d47d378315bdcad990620010c8452f3133d1) | [after](https://github.com/elastic/kibana/commit/a0235f861dd68ae902534a1537aedad74bea50cd) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/8263d47d378315bdcad990620010c8452f3133d1) | [after](https://github.com/elastic/kibana/commit/a0235f861dd68ae902534a1537aedad74bea50cd) | diff |
| --- | --- | --- | --- |
| `dashboard` | 200.3KB | 200.5KB | +277.0B |
| `discover` | 511.3KB | 511.4KB | +143.0B |
| total |  |  | +420.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/8263d47d378315bdcad990620010c8452f3133d1) | [after](https://github.com/elastic/kibana/commit/a0235f861dd68ae902534a1537aedad74bea50cd) | diff |
| --- | --- | --- | --- |
| `dashboard` | 334.2KB | 334.5KB | +325.0B |
| `data` | 1013.3KB | 1014.9KB | +1.6KB |
| `dataEnhanced` | 41.8KB | 41.3KB | -434.0B |
| total |  |  | +1.5KB |

### History
* :green_heart: [Build #100597](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/) succeeded 6385ef545d7cbc7cb5e0b343111bf5805d227617
* :green_heart: [Build #100393](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/) succeeded 75eebcaa1811b142c9485a49378a36608df67e54
* :broken_heart: [Build #100367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/) failed d01cade01ac63d0cf2ee55843709f69cf1faa548
* :green_heart: [Build #100309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/) succeeded 6cd417744d63642dc35903b0fec42ccf0b99f880
* :broken_heart: [Build #100278](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/) failed 3f3f71bb695cada935ff0c6c1c3eef18be8c083b

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100278/","number":"100278","commit":"3f3f71bb695cada935ff0c6c1c3eef18be8c083b"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/","number":"100309","commit":"6cd417744d63642dc35903b0fec42ccf0b99f880"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/","number":"100367","commit":"d01cade01ac63d0cf2ee55843709f69cf1faa548"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/","number":"100393","commit":"75eebcaa1811b142c9485a49378a36608df67e54"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/","number":"100597","commit":"6385ef545d7cbc7cb5e0b343111bf5805d227617"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100646/","number":"100646","commit":"a0235f861dd68ae902534a1537aedad74bea50cd"}]}
PIPELINE-->

kertal commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100998/)
* Commit: 377b76f285036204f14cd43673724be05992dfd5

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/f5e869189b0bbf242637a2fb8f06c1e5e74633b7) | [after](https://github.com/elastic/kibana/commit/377b76f285036204f14cd43673724be05992dfd5) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f5e869189b0bbf242637a2fb8f06c1e5e74633b7) | [after](https://github.com/elastic/kibana/commit/377b76f285036204f14cd43673724be05992dfd5) | diff |
| --- | --- | --- | --- |
| `dashboard` | 158.4KB | 158.7KB | +277.0B |
| `discover` | 420.2KB | 420.3KB | +143.0B |
| total |  |  | +420.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f5e869189b0bbf242637a2fb8f06c1e5e74633b7) | [after](https://github.com/elastic/kibana/commit/377b76f285036204f14cd43673724be05992dfd5) | diff |
| --- | --- | --- | --- |
| `dashboard` | 295.2KB | 295.5KB | +325.0B |
| `data` | 798.5KB | 799.2KB | +741.0B |
| `dataEnhanced` | 36.4KB | 36.0KB | -434.0B |
| total |  |  | +632.0B |

### History
* :green_heart: [Build #100646](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100646/) succeeded a0235f861dd68ae902534a1537aedad74bea50cd
* :green_heart: [Build #100597](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/) succeeded 6385ef545d7cbc7cb5e0b343111bf5805d227617
* :green_heart: [Build #100393](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/) succeeded 75eebcaa1811b142c9485a49378a36608df67e54
* :broken_heart: [Build #100367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/) failed d01cade01ac63d0cf2ee55843709f69cf1faa548
* :green_heart: [Build #100309](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/) succeeded 6cd417744d63642dc35903b0fec42ccf0b99f880

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100309/","number":"100309","commit":"6cd417744d63642dc35903b0fec42ccf0b99f880"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/","number":"100367","commit":"d01cade01ac63d0cf2ee55843709f69cf1faa548"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/","number":"100393","commit":"75eebcaa1811b142c9485a49378a36608df67e54"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/","number":"100597","commit":"6385ef545d7cbc7cb5e0b343111bf5805d227617"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100646/","number":"100646","commit":"a0235f861dd68ae902534a1537aedad74bea50cd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100998/","number":"100998","commit":"377b76f285036204f14cd43673724be05992dfd5"}]}
PIPELINE-->

kertal reviewed (approved):
Code LGTM, search sessions still work in Discover, tested locally in Chrome 👍

ThomThomson reviewed (approved):
Code only review - Dashboard changes LGTM.

Dosant commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101373/)
* Commit: 9b43a6e91000f5591f35f57dc2dc2dce721e388b

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/053628cae862103b8c27bc60656bf45d76dc3f5d) | [after](https://github.com/elastic/kibana/commit/9b43a6e91000f5591f35f57dc2dc2dce721e388b) | diff |
| --- | --- | --- | --- |
| `data` | 614 | 615 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/053628cae862103b8c27bc60656bf45d76dc3f5d) | [after](https://github.com/elastic/kibana/commit/9b43a6e91000f5591f35f57dc2dc2dce721e388b) | diff |
| --- | --- | --- | --- |
| `dashboard` | 158.4KB | 158.7KB | +277.0B |
| `discover` | 420.2KB | 420.3KB | +143.0B |
| total |  |  | +420.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/053628cae862103b8c27bc60656bf45d76dc3f5d) | [after](https://github.com/elastic/kibana/commit/9b43a6e91000f5591f35f57dc2dc2dce721e388b) | diff |
| --- | --- | --- | --- |
| `dashboard` | 295.2KB | 295.5KB | +325.0B |
| `data` | 798.3KB | 799.0KB | +741.0B |
| `dataEnhanced` | 36.4KB | 35.9KB | -434.0B |
| total |  |  | +632.0B |

### History
* :green_heart: [Build #100998](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100998/) succeeded 377b76f285036204f14cd43673724be05992dfd5
* :green_heart: [Build #100646](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100646/) succeeded a0235f861dd68ae902534a1537aedad74bea50cd
* :green_heart: [Build #100597](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/) succeeded 6385ef545d7cbc7cb5e0b343111bf5805d227617
* :green_heart: [Build #100393](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/) succeeded 75eebcaa1811b142c9485a49378a36608df67e54
* :broken_heart: [Build #100367](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/) failed d01cade01ac63d0cf2ee55843709f69cf1faa548

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100367/","number":"100367","commit":"d01cade01ac63d0cf2ee55843709f69cf1faa548"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100393/","number":"100393","commit":"75eebcaa1811b142c9485a49378a36608df67e54"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100597/","number":"100597","commit":"6385ef545d7cbc7cb5e0b343111bf5805d227617"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100646/","number":"100646","commit":"a0235f861dd68ae902534a1537aedad74bea50cd"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/100998/","number":"100998","commit":"377b76f285036204f14cd43673724be05992dfd5"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/101373/","number":"101373","commit":"9b43a6e91000f5591f35f57dc2dc2dce721e388b"}]}
PIPELINE-->
